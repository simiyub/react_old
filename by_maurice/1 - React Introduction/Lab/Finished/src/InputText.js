import React, { Component } from 'react';

class InputText extends Component {
  componentWillReceiveProps(newProps) {
    if (this.input.value !== newProps.value) {
      this.input.value = newProps.value;
    }
  }

  onChange = () => {
    const value = this.input.value;
    this.props.onChange({ value: value, prop: this.props.prop });
  };

  render() {
    return (
      <div className="form-group">
        <label>{this.props.children}</label>
        <input
          id={this.props.prop + 'Input'}
          type="text"
          className="form-control"
          ref={el => (this.input = el)}
          onChange={this.onChange}
        />
      </div>
    );
  }
}

export default InputText;
