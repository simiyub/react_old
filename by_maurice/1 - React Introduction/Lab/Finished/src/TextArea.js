import React, { Component } from 'react';

class TextArea extends Component {
  componentWillReceiveProps(newProps) {
    if (this.input.value !== newProps.value) {
      this.input.value = newProps.value;
    }
  }
  onChange = () => {
    const value = this.input.value;
    this.props.onChange({ value: value, prop: this.props.prop });
  };
  render() {
    return (
      <div className="form-group">
        <label>{this.props.children}</label>
        <textarea
          id={this.props.prop + 'Input'}
          className="form-control"
          ref={el => (this.input = el)}
          onChange={this.onChange}
          rows="5"
        />
      </div>
    );
  }
}

export default TextArea;
