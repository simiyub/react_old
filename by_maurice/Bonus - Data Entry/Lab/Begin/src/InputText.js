import React, { Component } from "react";
import PropTypes from "prop-types";

class InputText extends Component {
  static defaultProps = {
    value: ""
  };

  static propTypes = {
    children: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    prop: PropTypes.string.isRequired,
    value: PropTypes.any
  };

  render() {
    return (
      <div className="form-group">
        <label>{this.props.children}</label>
        <input
          id={this.props.prop + "Input"}
          type="text"
          className="form-control"
        />
      </div>
    );
  }
}

export default InputText;
