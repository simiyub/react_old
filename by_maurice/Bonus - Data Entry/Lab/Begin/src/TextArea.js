import React, { Component } from "react";
import PropTypes from "prop-types";

class TextArea extends Component {
  static defaultProps = {
    value: ""
  };

  static propTypes = {
    children: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    prop: PropTypes.string.isRequired,
    value: PropTypes.string
  };

  render() {
    return (
      <div className="form-group">
        <label>{this.props.children}</label>
        <textarea
          id={this.props.prop + "Input"}
          className="form-control"
          rows="5"
        />
      </div>
    );
  }
}

export default TextArea;
