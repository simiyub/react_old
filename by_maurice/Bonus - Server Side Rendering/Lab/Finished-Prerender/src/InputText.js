import React, { Component } from "react";

class InputText extends Component {
  static defaultProps = {
    value: ""
  };
  
  onChange = e => {
    const { value } = e.target;
    this.props.onChange({ value: value, prop: this.props.prop });
  };

  render() {
    const { value } = this.props;

    return (
      <div className="form-group">
        <label>{this.props.children}</label>
        <input
          type="text"
          className="form-control"
          value={value}
          onChange={this.onChange}
        />
      </div>
    );
  }
}

export default InputText;
