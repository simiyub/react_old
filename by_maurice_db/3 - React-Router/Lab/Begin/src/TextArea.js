import React, { Component } from 'react';

class TextArea extends Component {
    input = null;
    componentWillReceiveProps(newProps) {
      if (this.input.value !== newProps.value) {
        this.input.value = newProps.value;
      }
    }
    onChange = () => {
      var value = this.input.value;
      this.props.onChange({ value: value, prop: this.props.prop });
    };
    render() {
      return (
        <div className="form-group">
          <label>{this.props.children}</label>
          <textarea
            id={this.props.prop + "Input"}
            className="form-control"
            ref={e => (this.input = e)}
            onChange={this.onChange}
            rows="5"
          />
        </div>
      );
    }
  }
  
export default TextArea;