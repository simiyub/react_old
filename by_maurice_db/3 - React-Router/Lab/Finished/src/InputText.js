import React, { Component } from "react";

class InputText extends Component {
  input = null;

  componentWillReceiveProps(newProps) {
    if (this.input.value !== newProps.value) {
      this.input.value = newProps.value;
    }
  }
  onChange = () => {
    var value = this.input.value;
    this.props.onChange({ value: value, prop: this.props.prop });
  };
  render() {
    return (
      <div className="form-group">
        <label>{this.props.children}</label>
        <input
          id={this.props.prop + "Input"}
          type="text"
          className="form-control"
          ref={e => (this.input = e)}
          onChange={this.onChange}
        />
      </div>
    );
  }
}

export default InputText;
