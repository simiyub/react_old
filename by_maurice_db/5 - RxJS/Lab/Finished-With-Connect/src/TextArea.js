import React, { Component } from "react";

class TextArea extends Component {
  static defaultProps = {
    value: ""
  };

  onChange = e => {
    const { value } = e.target;
    this.props.onChange({ value: value, prop: this.props.prop });
  };

  render() {
    const { value } = this.props;

    return (
      <div className="form-group">
        <label>{this.props.children}</label>
        <textarea
          className="form-control"
          value={value}
          onChange={this.onChange}
          rows="5"
        />
      </div>
    );
  }
}

export default TextArea;
