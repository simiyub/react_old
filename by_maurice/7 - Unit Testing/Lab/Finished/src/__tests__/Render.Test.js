import React from "react";
import renderer from "react-test-renderer";

import MoviesList from "../MovieList";
import MovieEditor from "../MovieEditor";
import InputText from "../InputText";
import TextArea from "../TextArea";

describe("All components should render correctly", () => {
  it("empty MoviesList", () => {
    const tree = renderer
      .create(<MoviesList movies={[]} toEditMode={() => {}} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("non empty MoviesList", () => {
    const tree = renderer
      .create(
        <MoviesList
          movies={[
            {
              id: "1",
              title: "Kill Bill",
              abridgedDirectors: ["Quentin Tarantino"]
            }
          ]}
          toEditMode={() => {}}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("empty MovieEditor", () => {
    const tree = renderer
      .create(
        <MovieEditor
          movie={{
            id: "",
            title: "",
            abridgedDirectors: [],
            criticsConsensus: "",
            synopsis: "",
            year: 0
          }}
          save={() => {}}
          toListMode={() => {}}
          onChange={() => {}}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("non empty MovieEditor", () => {
    const tree = renderer
      .create(
        <MovieEditor
          movie={{
            id: "1",
            title: "Kill Bill",
            abridgedDirectors: ["Quentin Tarantino"],
            criticsConsensus: "Good",
            synopsis: "A revenge movie",
            year: 2003
          }}
          save={() => {}}
          toListMode={() => {}}
          onChange={() => {}}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("InputText", () => {
    const tree = renderer
      .create(
        <InputText value={"Kill Bill"} prop="title" onChange={() => {}}>
          Title
        </InputText>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("TextArea", () => {
    const tree = renderer
      .create(
        <TextArea
          value={"movie.criticsConsensus"}
          prop="criticsConsensus"
          onChange={() => {}}
        >
          Critics Consensus
        </TextArea>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
