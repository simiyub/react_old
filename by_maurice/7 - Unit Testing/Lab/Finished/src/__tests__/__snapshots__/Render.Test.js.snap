// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All components should render correctly InputText 1`] = `
<div
  className="form-group"
>
  <label>
    Title
  </label>
  <input
    className="form-control"
    onChange={[Function]}
    type="text"
    value="Kill Bill"
  />
</div>
`;

exports[`All components should render correctly TextArea 1`] = `
<div
  className="form-group"
>
  <label>
    Critics Consensus
  </label>
  <textarea
    className="form-control"
    id="criticsConsensusInput"
    onChange={[Function]}
    rows="5"
    value="movie.criticsConsensus"
  />
</div>
`;

exports[`All components should render correctly empty MovieEditor 1`] = `
<form>
  <div
    className="form-group"
  >
    <label>
      Title
    </label>
    <input
      className="form-control"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <label>
      Overview
    </label>
    <textarea
      className="form-control"
      id="overviewInput"
      onChange={[Function]}
      rows="5"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <label>
      Release date
    </label>
    <input
      className="form-control"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <label>
      popularity
    </label>
    <input
      className="form-control"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <label>
      vote average
    </label>
    <input
      className="form-control"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="btn-group"
  >
    <button
      className="btn btn-primary"
      onClick={[Function]}
      type="submit"
    >
      Save
    </button>
    <button
      className="btn btn-danger"
      onClick={[Function]}
    >
      Cancel
    </button>
  </div>
</form>
`;

exports[`All components should render correctly empty MoviesList 1`] = `
<table
  className="table table-bordered table-striped"
>
  <thead>
    <tr>
      <th>
        Title
      </th>
      <th />
    </tr>
  </thead>
  <tbody />
</table>
`;

exports[`All components should render correctly non empty MovieEditor 1`] = `
<form>
  <div
    className="form-group"
  >
    <label>
      Title
    </label>
    <input
      className="form-control"
      onChange={[Function]}
      type="text"
      value="Kill Bill"
    />
  </div>
  <div
    className="form-group"
  >
    <label>
      Overview
    </label>
    <textarea
      className="form-control"
      id="overviewInput"
      onChange={[Function]}
      rows="5"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <label>
      Release date
    </label>
    <input
      className="form-control"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <label>
      popularity
    </label>
    <input
      className="form-control"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <label>
      vote average
    </label>
    <input
      className="form-control"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="btn-group"
  >
    <button
      className="btn btn-primary"
      onClick={[Function]}
      type="submit"
    >
      Save
    </button>
    <button
      className="btn btn-danger"
      onClick={[Function]}
    >
      Cancel
    </button>
  </div>
</form>
`;

exports[`All components should render correctly non empty MoviesList 1`] = `
<table
  className="table table-bordered table-striped"
>
  <thead>
    <tr>
      <th>
        Title
      </th>
      <th />
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Kill Bill
      </td>
      <td
        style={
          Object {
            "width": 1,
          }
        }
      >
        <button
          className="btn btn-default btn-xs edit-button"
          onClick={[Function]}
        >
          Edit
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;
